apps/antalmanac/src/components/RightPane/SectionTable/EnrollmentHistoryPopup.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState, useEffect, useMemo } from 'react';
+import { LineChart, Line, CartesianGrid, ResponsiveContainer, XAxis, YAxis, Tooltip, Legend } from 'recharts';
+import { Box, Link, Typography, Skeleton } from '@mui/material';
+import enrollmentHistoryCache, { EnrollmentHistory } from '$lib/enrollmentHistory';
+import { isDarkMode } from '$lib/helpers';
+
+export interface EnrollmentHistoryPopupProps {
+    department: string;
+    courseNumber: string;
+    isMobileScreen: boolean;
+}
+
+const EnrollmentHistoryPopup = ({ department, courseNumber, isMobileScreen }: EnrollmentHistoryPopupProps) => {
+    const [loading, setLoading] = useState(true);
+    const [enrollmentHistory, setEnrollmentHistory] = useState<EnrollmentHistory>();
+
+    const graphWidth = useMemo(() => (isMobileScreen ? 250 : 450), [isMobileScreen]);
+    const graphHeight = useMemo(() => (isMobileScreen ? 175 : 250), [isMobileScreen]);
+    const popupTitle = useMemo(() => {
+        if (!enrollmentHistory) {
+            return 'No past enrollment data found for this course';
+        }
+
+        return `${department} ${courseNumber} | ${enrollmentHistory.year} ${
+            enrollmentHistory.quarter
+        } | ${enrollmentHistory.instructors.join(', ')}`;
+    }, [courseNumber, department, enrollmentHistory]);
+
+    const encodedDept = useMemo(() => encodeURIComponent(department), [department]);
+    const axisColor = isDarkMode() ? '#fff' : '#111';
+    const tooltipDateColor = '#111';
+
+    useEffect(() => {
+        if (!loading) {
+            return;
+        }
+
+        enrollmentHistoryCache.queryEnrollmentHistory(department, courseNumber).then((data) => {
+            if (data) {
+                setEnrollmentHistory(data);
+            }
+            setLoading(false);
+        });
+    }, [loading, department, courseNumber]);
+
+    if (loading) {
+        return (
+            <Box padding={1}>
+                <Skeleton variant="text" animation="wave" height={graphHeight} width={graphWidth} />
+            </Box>
+        );
+    }
+
+    if (!enrollmentHistory) {
+        return (
+            <Box padding={1}>
+                <Typography variant="body1" align="center">
+                    {popupTitle}
+                </Typography>
+            </Box>
+        );
+    }
+
+    return (
+        <Box sx={{ padding: '4px' }}>
```

Comment:
When assigning values, try to use raw numbers instead of specifying units. It's because those units will be used by MUI's theme awareness https://mui.com/system/getting-started/the-sx-prop/#spacing

```suggestion
        <Box sx={{ padding: 4}}>
```